packages/hooks/src/use_shared_state.rs

```diff
@@ -306,11 +309,26 @@ impl<T> Clone for UseSharedState<T> {
     }
 }
 
-impl<T: PartialEq> PartialEq for UseSharedState<T> {
-    fn eq(&self, other: &Self) -> bool {
-        let first = self.inner.borrow();
-        let second = other.inner.borrow();
-        first.value == second.value
+impl<T> UseFutureDep for &UseSharedState<T> {
```

Comment:
Can we make this the partialeq impl instead like use state does?

UseFutureDep is implemented for all tuples with partial eq elements. This may  fix your problem for a single dependency, but it will make use shared state unusable with multiple dependencies in a tuple

Reply:
OK this makes sense. I put a `gen` in `UseSharedState` itself to use in `PartialEq`, since the `gen`s in `ProvidedStateInner` will always be the same. I have them getting synced up in `use_shared_state`, similar to how `use_state` updates `current_val`. I'm not sure if this is the simplest solution, but it does seem to be working in tuples now.